### ISSUE TITLE
 Error: Failed to get data from GraphQL API for v2.5.0

### ISSUE THREAD

**Issue by karthikboncheruuvu** [advanced-security/policy-as-code#66]
HI @GeekMasher ...

We are seeing the below error in one of the repo when using v2.5.0, Kindly let us know if there is any solution for same.

Dependabot Results
  Error: Unknown Exception was hit, please repo this to https://github.com/advanced-security/policy-as-code
  Error: Failed to get data from GraphQL API

**Comment by GeekMasher** [comment]
My apologies @karthikboncheruuvu, very busy last 2 months. I'll add it to my backlog and work on this tomorrow.

**Comment by karthikboncheruuvu** [comment]
HI @GeekMasher ,
Thanks for looking back, hope i will get the update soon

**Comment by GeekMasher** [comment]
@karthikboncheruuvu Can you test this using the `update-depalerts` branch?

```yaml
- name: Advance Security Policy as Code
  uses: advanced-security/policy-as-code@update-depalerts
```

I have migrated to using the REST API for getting the alerts (but fallback to GraphQL) versus GraphQL. This should help with stability but would like you to test before the PR is merged #77

**Comment by karthikboncheruuvu** [comment]
No its still gives below error 

 Dependency Graph
  Error: Unknown Exception was hit, please repo this to https://github.com/advanced-security/policy-as-code
  Error: Failed to get data from GraphQL API

**Comment by karthikboncheruuvu** [comment]
I think its better to resolve all deps PRs raised by github which might solve many problems...
only my view!!! nothing specific

**Comment by GeekMasher** [comment]
@karthikboncheruuvu Sorry for the delay. I did some more testing and merged that PR. It should be supported in `v2.6`.

Is this related to #89 ?

**Comment by karthikboncheruuvu** [comment]
nope.. both are different.

**Comment by GeekMasher** [comment]
@karthikboncheruuvu Can you share the workflow file? and what permissions do you have set?

Recently I got an issue with a user around permissions. I just [updated the docs on this subject as it wasn't very clear](https://github.com/advanced-security/policy-as-code?tab=readme-ov-file#github-access-permissions).

**Comment by karthikboncheruuvu** [comment]
HI @GeekMasher :
i had given extra permission set as well.

permissions:
      contents: read
      id-token: write
      packages: write
      actions: read
      security-events: write
      
    issues resolved after using 2.7.0 version..
    thanks for responding and looking into it.

**Comment by GeekMasher** [comment]
Great to hear. It must have been some of the small changes. Closing issue now
